### PR TITLE
Bug 1131509 - [Flame][First Time Experience]SIM 2 PIN page's "<" button ...

### PR DESCRIPTION
**Pull request by mancas**
...can't be used in FTU. r=fcampo

### CODE REVIEW

**Comment by try-server-hook** [comment]
[mancas Manuel Casas (mancas)](https://github.com/mancas) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=cf09af048ce5)

File: apps/ftu/test/unit/sim_manager_test.js [hunk 1]
```diff
@@ -582,4 +582,46 @@ suite('sim mgmt >', function() {
     });
   });
 
+  suite('SIM2 inserted >', function() {
+    var iccId1;
+    var iccInfo1;
+    var realIccId0;
+
+    suiteSetup(function() {
```
Comment (fcampo) [inline apps/ftu/test/unit/sim_manager_test.js]: better to use `setup` so all the tests inside of the suite start fresh

File: apps/ftu/test/unit/sim_manager_test.js [hunk 2]
```diff
@@ -582,4 +582,46 @@ suite('sim mgmt >', function() {
...
+      navigator.mozMobileConnections[1].iccInfo = iccInfo1;
+
+      SimManager.updateIccState(iccId1);
+      SimManager.simSlots = 2;
+
+      realIccId0 = SimManager.icc0;
+      SimManager.icc0 = null;
+    });
+
+    suiteTeardown(function() {
```
Comment (fcampo) [inline apps/ftu/test/unit/sim_manager_test.js]: `teardown`

File: apps/ftu/test/unit/sim_manager_test.js [hunk 3]
```diff
@@ -582,4 +582,46 @@ suite('sim mgmt >', function() {
...
+      SimManager.simSlots = 1;
+      SimManager.icc1 = null;
+      SimManager.icc0 = realIccId0;
+    });
+
+    test('"simUnlockBack" hides the screen', function() {
+      iccInfo1.cardState = 'pinRequired';
+      SimManager.handleCardState();
+
+      navigationStub = sinon.stub(Navigation, 'back');
```
Comment (fcampo) [inline apps/ftu/test/unit/sim_manager_test.js]: if you use `this.sinon.stub(Navigation, 'back');` instead (inside of `setup`) you don't need to use `restore()` function

File: apps/ftu/test/unit/sim_manager_test.js [hunk 4]
```diff
@@ -582,4 +582,46 @@ suite('sim mgmt >', function() {
...
+      SimManager.simUnlockBack();
+      assert.ok(navigationStub.calledOnce);
```
Comment (fcampo) [inline apps/ftu/test/unit/sim_manager_test.js]: no need to name the stub, you can check for `assert.ok(Navigation.back.calledOnce);`

File: apps/ftu/test/unit/sim_manager_test.js [hunk 5]
```diff
@@ -582,4 +582,46 @@ suite('sim mgmt >', function() {
...
+      assert.isTrue(UIManager.activationScreen.classList.contains('show'));
+      assert.isFalse(UIManager.unlockSimScreen.classList.contains('show'));
+      assert.isTrue(UIManager.pinRetriesLeft.classList.contains('hidden'));
+      navigationStub.restore();
+    });
+  });
+
```
Comment (fcampo) [inline apps/ftu/test/unit/sim_manager_test.js]: we should also add a test checking that:
- SIM1 is PIN protected, but we skip it
- SIM2 is PIN protected, but we go back
- then SIM1 PIN screen is shown again (we are testing for hidden right now)

**Comment by try-server-hook** [comment]
[mancas Manuel Casas (mancas)](https://github.com/mancas) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8d4bfa404d04)

**Comment by try-server-hook** [comment]
[mancas Manuel Casas (mancas)](https://github.com/mancas) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=15cbcce3931d)

**Pull request merged**